### PR TITLE
[DataPipe] Adding a 's' to the functional names of open/list DataPipes

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #479

Differential Revision: [D36785643](https://our.internmc.facebook.com/intern/diff/D36785643)

### CODE REVIEW

File: torchdata/datapipes/iter/load/fsspec.py [hunk 1]
```diff
@@ -101,11 +101,11 @@ def __iter__(self) -> Iterator[str]:
                             yield abs_path
 
 
-@functional_datapipe("open_file_by_fsspec")
+@functional_datapipe("open_files_by_fsspec")
```
Comment (ejguan) [inline torchdata/datapipes/iter/load/fsspec.py:104]: I think this one is BC breaking from 0.3.0. Should we add `open_file_by_fsspec` to this class as well?

Call `IterDataPipe. register_datapipe_as_function("open_file_by_fsspec", FSSpecFileOpenerIterDataPipe)` at the end of this file.

Comment (NivekT) [reply torchdata/datapipes/iter/load/fsspec.py:104]: Yea, we can do that.

**Comment by NivekT** [comment]
> Thank you for adding this PR. Could you please also fix tests?

I don't think there are any existing use cases in tests but I will add some.

File: torchdata/datapipes/iter/load/fsspec.py [hunk 2]
```diff
@@ -133,6 +133,10 @@ def __len__(self) -> int:
         return len(self.source_datapipe)
 
 
+# Register for functional API for backward compatibility
+IterDataPipe.register_datapipe_as_function("open_file_by_fsspec", FSSpecFileOpenerIterDataPipe)
```
Comment (ejguan) [inline torchdata/datapipes/iter/load/fsspec.py:137]: As a follow-up task, add deprecation warning for the functional API.

Comment (NivekT) [reply torchdata/datapipes/iter/load/fsspec.py:137]: It seems like we will need to change the registration function and `IterDataPipe` class in PyTorch core to do that.  One possibility is [described here](https://github.com/pytorch/data/issues/480#issuecomment-1142500688).

Do you think we should land this PR as it is or make the changes to core first?

Comment (ejguan) [reply torchdata/datapipes/iter/load/fsspec.py:137]: Let's land this PR first and change the core afterwards.

**Review by ejguan — APPROVED** [review]

File: torchdata/datapipes/iter/load/fsspec.py (hunk 2 above)
Comment (ejguan) [reply torchdata/datapipes/iter/load/fsspec.py:137]: And, could you please update the deprecation table as well?

**Pull request closed without merging**